storage/core/src/main/java/io/aiven/kafka/tieredstorage/storage/proxy/ProxyConfig.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.storage.proxy;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+public class ProxyConfig extends AbstractConfig {
+    public static final String PROXY_PREFIX = "proxy.";
+
+    private static final String PROXY_HOST = "host";
+    private static final String PROXY_HOST_DOC = "The SOCKS5 proxy host name or IP address";
```

Comment:
Seems this config class is meant to be generic. Should we remove SOCKS5 mentions? Otherwise, maybe we could call it Socks5ProxyConfig or similar?

Reply:
My thinking:
1. Make the class generic.
2. Support only SOCKS5 at the moment.
3. If need be to introduce e.g. HTTP proxying, we add the `proxy.type` parameter.